.github/workflows/docs.yml

```
@@ -0,0 +1,43 @@
+name: CI
+on:
+  push:
+    branches:
+      - "master"
+
+jobs:
+  deploy-docs:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+      pages: write
+      id-token: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install mdbook
+        run: |
+          tag=$(curl 'https://api.github.com/repos/rust-lang/mdbook/releases/latest' | jq -r '.tag_name')
+          url="https://github.com/rust-lang/mdbook/releases/download/${tag}/mdbook-${tag}-x86_64-unknown-linux-gnu.tar.gz"
+          mkdir mdbook
+          curl -sSL $url | tar -xz --directory=./mdbook
+          echo `pwd`/mdbook >> $GITHUB_PATH
```

Micro-optimising bash scripts is my passion.
```suggestion
          curl -sSL $url | tar -xz -C mdbook
          printf '%s/mdbook\n' "$PWD" >> $GITHUB_PATH
```